runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -348,8 +373,57 @@ decl_module! {
         /// Exports const -  max simultaneous active proposals number.
         const MaxActiveProposalLimit: u32 = T::MaxActiveProposalLimit::get();
 
+        /// Block Initialization. Perform voting period check, vote result tally, approved proposals
+        /// grace period checks, and proposal execution.
+        fn on_initialize() -> Weight {
+            // `process_proposal` returns the weight of the executed proposals. The weight of the
+            // executed proposals doesn't include any access to the store or calculation that
+            // `on_initialize` does. Therefore, to get the total weight of `on_initialize` we need
+            // to add the weight of the execution of `on_intialize` to the weight returned by
+            // `process_proposal`.
+            // To be safe, we use the worst possible case for `on_initialize`, meaning that there
+            // are as many proposals active as possible and they all take the worst possible branch.
+
+            let max_active_proposals = T::MaxActiveProposalLimit::get();
+
+            // Weight when all the proposals are immediatly approved and executed
+            let immediate_execution_branch_weight =
```

Comment:
Much better now. But I see the possibility to move the weight-code to a separate function. Do you think it would be better?

Reply:
I agree! moved most of the calculation to a separate function.